.pre-commit-config.yaml

```diff
@@ -19,19 +9,18 @@ repos:
       - id: trailing-whitespace
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.7.4
+    rev: v0.11.4
     hooks:
       - id: ruff
+        args: [ --fix ]
       - id: ruff-format
 
-  - repo: local
+  - repo: https://github.com/tox-dev/pyproject-fmt
+    rev: v2.5.1
     hooks:
-      - id: pytest
-        name: Run pytest
-        entry: make test
-        language: system
-        pass_filenames: false
-        always_run: true
+      - id: pyproject-fmt
 
-ci:
-  skip: [pytest]
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.24.1
```

Comment:
What's the reasoning behind using both of these? Looking at their documentation, they are very similar.

Reply:
One formats the other validates. Sometimes the formatter can mess the file up, the validator prevents it from it getting commited.

Reply:
Oh, I see, it makes sense! Thanks for the clarification!